wps/core/repository.py

```diff
@@ -250,3 +251,26 @@ async def get_dataset(self, dataset_id: str) -> Dataset:
             return await self._dataset_dao.get_by_id(dataset_id)
         except ResourceNotFoundError as error:
             raise self.DatasetNotFoundError("Dataset not found") from error
+
+    async def get_datasets(
+        self, *, auth_context: AuthContext, work_type: Optional[WorkType] = None
```

Comment:
Allowing None as work type filter was intentational, so that the frontend can get both downloadable and uploadable datasets with one call (e.g. for the profile page).

Reply:
Ah OK, that's all right. Maybe can we point that out in the doc string a bit more?

Reply:
Done in 42a3351.